core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -161,47 +162,47 @@ class HttpGoogleServicesDAO(
 
-    def insertBucket: Map[WorkspaceAccessLevel, WorkbenchEmail] => Future[String] = { policyGroupsByAccessLevel =>
+    def insertBucket(): Future[String] = {
       implicit val service = GoogleInstrumentedService.Storage
-      retryWhen500orGoogleError {
-        () => {
-          // bucket ACLs should be:
-          //   project owner - bucket writer
-          //   workspace owner - bucket writer
-          //   workspace writer - bucket writer
-          //   workspace reader - bucket reader
-          //   bucket service account - bucket owner
-          val workspaceAccessToBucketAcl: Map[WorkspaceAccessLevel, String] = Map(ProjectOwner -> "WRITER", Owner -> "WRITER", Write -> "WRITER", Read -> "READER")
-          val bucketAcls =
-            policyGroupsByAccessLevel.map { case (access, policyEmail) => newBucketAccessControl(makeGroupEntityString(policyEmail.value), workspaceAccessToBucketAcl(access)) }.toSeq :+
-              newBucketAccessControl("user-" + clientEmail, "OWNER")
-
-          // default object ACLs should be:
-          //   project owner - object reader
-          //   workspace owner - object reader
-          //   workspace writer - object reader
-          //   workspace reader - object reader
-          //   bucket service account - object owner
-          val defaultObjectAcls =
-          policyGroupsByAccessLevel.map { case (_, policyEmail) => newObjectAccessControl(makeGroupEntityString(policyEmail.value), "READER") }.toSeq :+
-              newObjectAccessControl("user-" + clientEmail, "OWNER")
-
-          val logging = new Logging().setLogBucket(getStorageLogsBucketName(project.projectName))
-
-          val bucket = new Bucket().
-            setName(bucketName).
-            setAcl(bucketAcls.asJava).
-            setDefaultObjectAcl(defaultObjectAcls.asJava).
-            setLogging(logging).
-            setLabels(labels.asJava)
-          val inserter = getStorage(getBucketServiceAccountCredential).buckets.insert(project.projectName.value, bucket)
-          executeGoogleRequest(inserter)
-
-          bucketName
-        }
+
+      val logging = new Logging().setLogBucket(getStorageLogsBucketName(project.projectName))
+      val bucket = new Bucket().setName(bucketName).setLogging(logging).setLabels(labels.asJava)
+
+      val inserter = getStorage(getBucketServiceAccountCredential).buckets.insert(project.projectName.value, bucket)
+
+      retryWhen500orGoogleError { () =>
+        executeGoogleRequest(inserter)
+        bucketName
       }
     }
 
+    def updateBucketIam: Map[WorkspaceAccessLevel, WorkbenchEmail] => Stream[IO, Unit] = { policyGroupsByAccessLevel =>
+      //default object ACLs are no longer used. bucket only policy is enabled on buckets to ensure that objects
+      //do not have separate permissions that deviate from the bucket-level permissions.
+      //
+      // project owner - roles/storage.objectAdmin
```

Comment:
I don't think we want to grant objectAdmin as that includes iam stuff. Replace objectAdmin with both objectCreator and objectViewer. But I think you said you also need `storage.objects.update`, oh well, I think you need a custom role.